identity-iota-core/src/document/iota_document.rs

```diff
@@ -296,43 +296,28 @@ impl IotaDocument {
     Ok(self.document.detach_method_relationship(did_url, relationship)?)
   }
 
-  /// Returns the first [`IotaVerificationMethod`] with an `id` property
-  /// matching the provided `query`.
-  pub fn resolve_method<'query, Q>(&self, query: Q) -> Option<&IotaVerificationMethod>
+  /// Returns the first [`IotaVerificationMethod`] with an `id` property matching the
+  /// provided `query` and the verification relationship specified by `scope` if present.
+  pub fn resolve_method<'query, Q>(&self, query: Q, scope: Option<MethodScope>) -> Option<&IotaVerificationMethod>
   where
     Q: Into<DIDUrlQuery<'query>>,
   {
-    self.document.resolve_method(query)
-  }
-
-  /// Returns the first [`IotaVerificationMethod`] with an `id` property
-  /// matching the provided `query`.
-  ///
-  /// # Errors
-  ///
-  /// Fails if no matching verification [`IotaVerificationMethod`] is found.
-  pub fn try_resolve_method<'query, Q>(&self, query: Q) -> Result<&IotaVerificationMethod>
-  where
-    Q: Into<DIDUrlQuery<'query>>,
-  {
-    Ok(self.document.try_resolve_method(query)?)
+    self.document.resolve_method(query, scope)
   }
 
   #[doc(hidden)]
-  pub fn try_resolve_method_mut<'query, Q>(&mut self, query: Q) -> Result<&mut IotaVerificationMethod>
+  pub fn resolve_method_mut<'query, Q>(
```

Comment:
This no longer needs `#[doc(hidden)]` since developers can no longer violate unsafe restrictions using it as far as I'm aware. It needs a proper doc-comment and warning though, since changing the id fragments may violate the uniqueness of verification methods.